lib/defopt.py

```diff
@@ -196,14 +211,21 @@ def run(funcs: Union[Callable, List[Callable], Dict[str, Callable]], *,
         dashes) to letters, to use as alternative short flags.  Defaults to
         `None`, which means to generate short flags for any non-ambiguous
         option.  Set to ``{}`` to completely disable short flags.
+    :param cli_options:
+        If ``all``, all parameters are converted into command-line flags.  If
+        ``has_default``, all parameters with a default are converted into
+        command-line flags.  The default behavior (``kwonly``) is to convert
+        keyword-only parameters to command line flags, and non-keyword-only
+        parameters with a default to optional positional command line
+        parameters.
```

Comment:
I think I prefer reordering to start with the default ("The default behavior ('kwonly') is ....  'all' turns all parameters into command-line flags; 'has_default' turns a parameter into a command-line flag if and only if it has a default value.")